packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -421,12 +360,9 @@ export default function App() {
             {...rainbowKitProviderProps}
           >
             <ConnectedChainSyncer />
-            <WelcomeDialog {...welcomeDialogProps} onClose={onClose} />
-            <NetworkReady>
-              <AppContextProvider>
-                <Injector>{isTosAccepted && <AppContent />}</Injector>
-              </AppContextProvider>
-            </NetworkReady>
+            <AppContextProvider>
+              <AppContent />
+            </AppContextProvider>
```

Comment:
combined the UI logic in `NetworkReady`, `Injector`, and `AppContent` into `AppContent`

Reply:
let's keep it as is, but don't think this was necessary, because now we'll have to start a few of our other PRs from scratch due to merge conflicts (like not requiring wallet connection) and imo we haven't gained much

Reply:
i'll go and fix the merge conflicts in those PRs if needed

we had a few trying to refactor it to fix the rerendering which causes e2es to be unstable

here when i update the UI it was very difficult to reorganise the UI without this change, and it's during this change that i could confirm some components and code paths had been unreachable for months